### PR TITLE
Support for Graylog 1.1

### PR DESCRIPTION
**Pull request by linuxprofessor**
Support for Graylog 1.1

* Changed variable mongodb_host to mongodb_hosts as it will now contain one or more hosts
* Added support for MongoDB URI instead of legacy MongoDB configuration method
* Removed the need for mongodb_port as this will be included in mongodb_hosts
* Changed development env to support Graylog 1.1 and ElasticSearch 1.5.2

### CODE REVIEW

File: graylog2 [hunk 1]
```diff
@@ -0,0 +1 @@
+graylog2
```
Comment (bernd) [inline graylog2]: What is this?

**Comment by bernd** [comment]
@linuxprofessor Thank you very much for the PR.

I would like to keep backwards compatibility with 1.0 and suggest to keep the `mongodb_host` and `mongodb_port` options.

We can build the new `mongodb_hosts` option from them if they are in use. Also, when they are in use we can log a warning that these options are deprecated.

What do you think?

File: graylog2 (hunk 1 above)
Comment (linuxprofessor) [inline graylog2]: That is a mistake, I'll remove it.

**Comment by linuxprofessor** [comment]
Could do that, yes. Is there a good way to determine which Graylog version that's used without the user specifying it? If graylog2::repo is used it's trivial, but I don't suppose one can count on everyone using it?

**Comment by linuxprofessor** [comment]
I've re-added support for v1.0 and updated README.md a bit.

Thinking about mongodb_replica_set, as it has totally different meanings if you're running the deprecated configuration method or the newer mongodb_uri method. Right now the content of that variable should be a number of hosts/ports in the old method and a replica set name in the new one. Can I leave it like this, or should they be separated?

File: README.md [hunk 2]
```diff
@@ -91,11 +91,11 @@ class {'graylog2::web':
 }
 ```
 
-Installing graylog-radio with default settings
+Installing graylog-radio with default settings (deprecated since Graylog 1.0)
 
 ```puppet
 class {'graylog2::repo':
-  version => '1.0'
+  version => '0.92'
```
Comment (bernd) [inline README.md]: I guess this should be `1.1`. I can fix that after merging.

Comment (linuxprofessor) [inline README.md]: I changed it to 0.92 since I was under the impression that graylog radio isn't supported after 1.0

**Comment by bernd** [comment]
Thanks for the updates. Good point with the replica set setting. I will think about that.

File: README.md (hunk 2 above)
Comment (bernd) [inline README.md]: It's deprecated but it still works in 1.x.

**Comment by bernd** [comment]
@linuxprofessor Thanks for you hard work! I merged it and will test a bit and then decide what to do with the replica set issue.

**Comment by bernd** [comment]
@linuxprofessor What do you think about just adding a `mongodb_uri` config option and deprecating the rest?

**Comment by linuxprofessor** [comment]
@bernd that would be alright, can prior versions handle ```mongodb_uri```?

**Comment by bernd** [comment]
> that would be alright, can prior versions handle `mongodb_uri` ?

No, older versions don't know about that. My idea is as follows.

Using `mongodb_uri` will add the `mongodb_uri` setting to the server config.

```puppet
class { 'graylog::server':
  mongodb_uri => 'mongodb://localhost/graylog',
}
```

Using the old options like `mongodb_database` and `mongodb_host` etc. will just write those (deprecated) options to the server config.

```puppet
class { 'graylog::server':
  mongodb_database => 'graylog',
  mongodb_host     => 'localhost',
}
```

Usage of any of the deprecated options will trigger a warning in the puppet log.

**Comment by linuxprofessor** [comment]
Go for it, seems reasonable.

**Pull request merged**